### PR TITLE
Feature/book details

### PR DESCRIPTION
**Pull request by GCerbelli**
## Summary

- Develop book details feature

## Trello Card

[Nº 3.2 Detalle del libro](https://trello.com/c/CB35SFic/17-n-32-detalle-del-libro)

## Screenshots
BookDetails

### CODE REVIEW

File: wbooks/src/app/components/BookDetail/index.tsx [hunk 1]
```diff
@@ -1,8 +1,39 @@
 import React from 'react';
-import { Text } from 'react-native';
+import { View, Text, Image } from 'react-native';
 
-const BookDetail = () => {
-  return <Text>Book Detail</Text>;
+import styles from './styles';
```
Comment (guidoprinc) [inline wbooks/src/app/components/BookDetail/index.tsx:4]: siblings files should be separated from parents files. Also parents files should be next to libraries. So move them up

File: wbooks/src/app/components/BookDetail/styles.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    fontSize: 16,
+    marginTop: 5,
+    fontWeight: 'bold',
+    color: available
+  },
+  statusNotAvailable: {
+    fontSize: 16,
+    marginTop: 5,
+    fontWeight: 'bold',
+    color: notAvailable
```
Comment (guidoprinc) [inline wbooks/src/app/components/BookDetail/styles.ts:46]: they have 3 same values. create a common style and only change color for availability

File: wbooks/src/app/components/baseComponents/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+const Button = ({ text, fill, onPress, disabled }: ButtonProps) => {
+  let buttonStyle = {};
+  if (disabled) {
+    buttonStyle = [styles.button, styles.disabledButton];
+  } else if (fill) {
+    buttonStyle = [styles.button, styles.fillButton];
+  } else {
+    buttonStyle = styles.button;
+  }
```
Comment (guidoprinc) [inline wbooks/src/app/components/baseComponents/Button/index.tsx:15]: avoid doing this and only do a conditional inside the styles

File: wbooks/src/app/components/baseComponents/Button/styles.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+const styles = StyleSheet.create({
+  button: {
+    height: 45,
+    borderRadius: 20,
+    marginTop: 10,
+    borderWidth: 1.5,
+    justifyContent: 'center',
+    alignItems: 'center',
+    flexDirection: 'column',
```
Comment (guidoprinc) [inline wbooks/src/app/components/baseComponents/Button/styles.ts:13]: by default it should be column

File: wbooks/src/app/components/BookDetail/index.tsx [hunk 5]
```diff
@@ -1,8 +1,38 @@
...
+  return (
+    <View style={styles.background}>
+      <View style={styles.container}>
+        <View style={styles.card}>
+          <Image style={styles.image} source={image} />
+          <View style={styles.infoContainer}>
+            <Text numberOfLines={1} style={styles.title}>
+              {title}
+            </Text>
+            <Text style={available ? [styles.status, styles.available] : styles.status}>
```
Comment (marianozicavo) [inline wbooks/src/app/components/BookDetail/index.tsx:23]:
```suggestion
            <Text style={[styles.status, available && styles.available]}>
```

File: wbooks/src/app/components/BookDetail/styles.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { StyleSheet } from 'react-native';
+
+import { lightBlue, white, dark, available, notAvailable } from '../../../constants/colors';
+
+const styles = StyleSheet.create({
+  background: {
+    backgroundColor: lightBlue,
+    padding: 15,
+    height: '100%'
```
Comment (marianozicavo) [inline wbooks/src/app/components/BookDetail/styles.ts:9]: Use `flex: 1` instead

File: wbooks/src/app/components/BookDetail/styles.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    height: 120,
+    width: 80
+  },
+  card: {
+    flexDirection: 'row',
+    marginBottom: 20
+  },
+  infoContainer: {
+    marginLeft: 20,
+    flexDirection: 'column'
```
Comment (marianozicavo) [inline wbooks/src/app/components/BookDetail/styles.ts:30]: this is the default value so you don't need it

File: wbooks/src/app/components/BookDetail/styles.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+const styles = StyleSheet.create({
+  background: {
+    backgroundColor: lightBlue,
+    padding: 15,
+    height: '100%'
+  },
+  container: {
+    backgroundColor: white,
+    borderRadius: 5,
+    flexDirection: 'column',
```
Comment (marianozicavo) [inline wbooks/src/app/components/BookDetail/styles.ts:14]: this is the default value so you don't need it

File: wbooks/src/app/components/baseComponents/Button/index.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+import { ButtonProps } from '../../../interfaces/button';
+import styles from './styles';
+
+const Button = ({ text, fill, onPress, disabled }: ButtonProps) => (
+  <TouchableOpacity
+    style={[styles.button, fill && styles.fillButton, disabled && styles.disabledButton]}
+    disabled={disabled}
+    onPress={onPress}>
+    <Text style={fill ? [styles.title, styles.fillButtonTitle] : styles.title}>{text}</Text>
```
Comment (marianozicavo) [inline wbooks/src/app/components/baseComponents/Button/index.tsx:12]:
```suggestion
    <Text style={[styles.title, fill && styles.fillButtonTitle]}>{text}</Text>
```

File: wbooks/src/app/components/baseComponents/Button/index.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
```
Comment (marianozicavo) [inline wbooks/src/app/components/baseComponents/Button/index.tsx:1]: Move this file to `src/app/components/Button/index.tsx`

File: wbooks/src/app/components/baseComponents/Button/index.tsx [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import { Text, TouchableOpacity } from 'react-native';
+
+import { ButtonProps } from '../../../interfaces/button';
```
Comment (marianozicavo) [inline wbooks/src/app/components/baseComponents/Button/index.tsx:4]: We declare the props in the same file. It should be 

````
interface Props {
  blabla
}


const Button = ({ text, fill, onPress, disabled }: Props) => (
```

File: wbooks/src/app/components/Button/index.tsx [hunk 12]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { Text, TouchableOpacity } from 'react-native';
+
+import styles from './styles';
+
+interface ButtonProps {
+  text: string;
+  fill?: boolean;
+  onPress?: any;
```
Comment (guidoprinc) [inline wbooks/src/app/components/Button/index.tsx:9]: you can define a function like `() => void`

File: wbooks/src/app/components/Button/index.tsx [hunk 13]
```diff
@@ -1,9 +1,15 @@
 import React from 'react';
 import { Text, TouchableOpacity } from 'react-native';
 
-import { ButtonProps } from '../../../interfaces/button';
 import styles from './styles';
 
+interface ButtonProps {
```
Comment (marianozicavo) [inline wbooks/src/app/components/Button/index.tsx:6]: We name these `Props` if they are used only here

File: wbooks/src/app/screens/BookDetail/styles.ts [hunk 14]
```diff
@@ -0,0 +1,49 @@
+import { StyleSheet } from 'react-native';
+
+import { lightBlue, white, dark, available, notAvailable } from '../../../constants/colors';
```
Comment (guidoprinc) [inline wbooks/src/app/screens/BookDetail/styles.ts:3]: we should always use color names to have an idea of it and not to be confused. So we need to change `available` and `notAvailable` for something more properly

Comment (GCerbelli) [reply wbooks/src/app/screens/BookDetail/styles.ts:3]: Ok i'll change it in next PR, thanks!

**Pull request merged**